core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -1982,25 +1982,8 @@ class GroupMetadataManagerTest {
     // expire all of the offsets
     time.sleep(4)
 
-    // expect the offset tombstone
-    val recordsCapture: ArgumentCaptor[MemoryRecords] = ArgumentCaptor.forClass(classOf[MemoryRecords])
-
-    when(partition.appendRecordsToLeader(recordsCapture.capture(),
-      origin = ArgumentMatchers.eq(AppendOrigin.COORDINATOR), requiredAcks = anyInt(),
-      any(), any())).thenReturn(LogAppendInfo.UNKNOWN_LOG_APPEND_INFO)
     groupMetadataManager.cleanupGroupMetadata()
 
-    // verify the tombstones are correct and only for the expired offsets
-    val records = recordsCapture.getValue.records.asScala.toList
-    assertEquals(2, records.size)
-    records.foreach { message =>
-      assertTrue(message.hasKey)
-      assertFalse(message.hasValue)
-      val offsetKey = GroupMetadataManager.readMessageKey(message.key).asInstanceOf[OffsetKey]
-      assertEquals(groupId, offsetKey.key.group)
-      assertEquals("foo", offsetKey.key.topicPartition.topic)
-    }
```

Comment:
Why is this no longer needed?

Reply:
I added back this code and made the test pass by returning the right partition from replica manager.